src/tyr/route_serializer_osrm.cc

```diff
@@ -33,6 +35,86 @@ struct NamedSegment {
   float distance;
 };
 
+const constexpr double TILE_SIZE = 256.0;
+static constexpr unsigned MAX_ZOOM = 18;
+static constexpr unsigned MIN_ZOOM = 1;
+// this is an upper bound to current display sizes
+static constexpr double VIEWPORT_WIDTH = 8 * TILE_SIZE;
+static constexpr double VIEWPORT_HEIGHT = 5 * TILE_SIZE;
+static double INV_LOG_2 = 1. / std::log(2);
+const constexpr double DEGREE_TO_RAD = 0.017453292519943295769236907684886;
+const constexpr double RAD_TO_DEGREE = 1. / DEGREE_TO_RAD;
+const constexpr double EPSG3857_MAX_LATITUDE = 85.051128779806592378; // 90(4*atan(exp(pi))/pi-1)
+
+const constexpr float DOUGLAS_PEUCKER_THRESHOLDS[19] = {
+    49438476562500.f, // z0
+    12359619140625.f, // z1
+    3089903027344.f,  // z2
+    772475756836.f,   // z3
+    193118939209.f,   // z4
+    48279515076.f,    // z5
+    12069878769.f,    // z6
+    3017414761.f,     // z7
+    754326225.f,      // z8
+    188567824.f,      // z9
+    47141956.f,       // z10
+    11785489.f,       // z11
+    2944656.f,        // z12
+    736164.f,         // z13
+    184041.f,         // z14
+    45796.f,          // z15
+    11449.f,          // z16
+    2809.f,           // z17
+    676.f,            // z18
+};
```

Comment:
our generalization works with units of `meters` since that is what `PointLL::Distance` returns. these appear to be something wholly different. we need to figure out what unit this is in. it doesnt look like meters per mercator pixel either...

Reply:
so i thought this was mercator pixels squared and wrote a little pythong thing to check it...

```python
#!/usr/bin/env python3
import math
simp = [49438476562500,
12359619140625,
3089903027344,
772475756836,
193118939209,
48279515076,
12069878769,
3017414761,
754326225,
188567824,
47141956,
11785489,
2944656,
736164,
184041,
45796,
11449,
2809,
676]

for i, s in enumerate(simp):
  mpp = 40041472 / (2**i * 256)
  p = math.sqrt(s)
  print("%.1f * %.3f = %.1f" % (p, mpp, p * mpp))
```

which gives us the final meters numbers of:

```
7031250.0 * 156412.000 = 1099771875000.0
3515625.0 * 78206.000 = 274942968750.0
1757812.0 * 39103.000 = 68735722636.0
878906.0 * 19551.500 = 17183930659.0
439453.0 * 9775.750 = 4295982664.8
219726.0 * 4887.875 = 1073993222.2
109863.0 * 2443.938 = 268498305.6
54931.0 * 1221.969 = 67123965.4
27465.0 * 610.984 = 16780685.9
13732.0 * 305.492 = 4195018.7
6866.0 * 152.746 = 1048754.7
3433.0 * 76.373 = 262188.7
1716.0 * 38.187 = 65528.1
858.0 * 19.093 = 16382.0
429.0 * 9.547 = 4095.5
214.0 * 4.773 = 1021.5
107.0 * 2.387 = 255.4
53.0 * 1.193 = 63.2
26.0 * 0.597 = 15.5
```
The numbers here are still way too big... 

Having a look at this stuff:
https://github.com/Project-OSRM/osrm-backend/blob/82b5648c97edf1d2edec7aecebc35aa8a8033c82/src/engine/douglas_peucker.cpp#L81
https://github.com/Project-OSRM/osrm-backend/blob/82b5648c97edf1d2edec7aecebc35aa8a8033c82/include/util/web_mercator.hpp#L131

I can only tell that the units are squared. I still cant tell what the units are exactly so i have to google what this conversion is.

@danpat if you have any clue what `util::web_mercator::fromWGS84` projects the polar coordinate to, that would be awesome to know :smile:

Reply:
```python
#!/usr/bin/env python3
import math
simp = [49438476562500,
12359619140625,
3089903027344,
772475756836,
193118939209,
48279515076,
12069878769,
3017414761,
754326225,
188567824,
47141956,
11785489,
2944656,
736164,
184041,
45796,
11449,
2809,
676]

for i, s in enumerate(simp):
  mpp = 40041472 / (2**i * 256)
  p = math.sqrt(s)
  print("%.1f * %.3f = %.1f" % (p, mpp, p * mpp))
```
so locking it to z18 meters per pixel gives the following values which are much more sane:

```
7031250.0 * 0.597 = 4195296.8
3515625.0 * 0.597 = 2097648.4
1757812.0 * 0.597 = 1048823.9
878906.0 * 0.597 = 524411.9
439453.0 * 0.597 = 262206.0
219726.0 * 0.597 = 131102.7
109863.0 * 0.597 = 65551.3
54931.0 * 0.597 = 32775.4
27465.0 * 0.597 = 16387.4
13732.0 * 0.597 = 8193.4
6866.0 * 0.597 = 4096.7
3433.0 * 0.597 = 2048.3
1716.0 * 0.597 = 1023.9
858.0 * 0.597 = 511.9
429.0 * 0.597 = 256.0
214.0 * 0.597 = 127.7
107.0 * 0.597 = 63.8
53.0 * 0.597 = 31.6
26.0 * 0.597 = 15.5
```

Reply:
i was thinking perhaps its mercator meters squared? but 26 is still too big for zl18. perhaps its mercator decimeters squared? which would make zl18 2.6 meters zl17 5.3m and so on.. that seems more reasonable, lets try that and see how it looks..